tests/AMSlib/cpu_packing_test.cpp

```diff
@@ -129,38 +141,45 @@ int main(int argc, char* argv[])
                   << ")\n";
         return 1;
       }
-
-      rm.copy(dense, h_dense);
-
-      if (verify(h_dense, elements, flag)) {
-        std::cout << "Dense elements do not have the correct values\n";
-        return 1;
+  
+      for ( auto d : d_data ){
+        rm.copy(d, h_dense);
+        if (verify(h_dense, elements, flag)) {
+          std::cout << "Dense elements do not have the correct values\n";
+          return 1;
+        }
       }
 
       data_handler::unpack(resource, predicate, size, d_data, sr_data, flag);
 
-      rm.copy(rsparse, h_rsparse);
-
-      if (verify(h_predicate, h_sparse, h_rsparse, size, flag)) {
-        //      for ( int k = 0; k < SIZE; k++){
-        //        std::cout << k << " " << h_sparse[k] << " " << h_rsparse[k] <<
-        //        "\n";
-        //      }
-        std::cout << "Unpacking packed data does not match initial values\n";
-        return 1;
+      for ( int i = 0; i < dims; i++){
+        rm.copy(sr_data[i], h_rsparse);
+        rm.copy(const_cast<double*>(s_data[i]), h_sparse);
+        if (verify(h_predicate, h_sparse, h_rsparse, size, flag)) {
+           std::cout << "Unpacking packed data does not match initial values\n";
+          return 1;
+        }
       }
     }
 
-    rm.deallocate(predicate, AMSResourceType::DEVICE);
     rm.deallocate(h_predicate, AMSResourceType::HOST);
-    rm.deallocate(dense, AMSResourceType::DEVICE);
     rm.deallocate(h_dense, AMSResourceType::HOST);
-    rm.deallocate(sparse, AMSResourceType::DEVICE);
     rm.deallocate(h_sparse, AMSResourceType::HOST);
-    rm.deallocate(rsparse, AMSResourceType::DEVICE);
     rm.deallocate(h_rsparse, AMSResourceType::HOST);
-    rm.deallocate(reindex, AMSResourceType::DEVICE);
   }
 
+  rm.deallocate(predicate, resource);
+  for (int i = 0; i < dims; i++){
+    rm.deallocate(const_cast<double*>(s_data[i]), resource);
+    rm.deallocate(sr_data[i], resource);
+    rm.deallocate(d_data[i], resource);
+  }
+  s_data.clear();
+  sr_data.clear();
+  d_data.clear();
+  cudaDeviceReset();
```

Comment:
This call `cudaDeviceReset()` to seems to be problematic on the CI side but works fine on LC machines for some reasons. We can just add:
```C
#include "cuda.h"
#include "cuda_runtime.h"
```

Reply:
I think I should guard it with `ENABLE_CUDA`. Will do.